controllers/gatekeeper_controller.go

```diff
@@ -201,16 +216,50 @@ func (r *GatekeeperReconciler) deployGatekeeperResources(gatekeeper *operatorv1a
 }
 
 func getStaticAssets(gatekeeper *operatorv1alpha1.Gatekeeper) []string {
-	if gatekeeper.Spec.ValidatingWebhook == nil || *gatekeeper.Spec.ValidatingWebhook == operatorv1alpha1.WebhookEnabled {
+	validatingWebhookEnabled := gatekeeper.Spec.ValidatingWebhook == nil || *gatekeeper.Spec.ValidatingWebhook == operatorv1alpha1.WebhookEnabled
+	mutatingWebhookEnabled := mutatingWebhookEnabled(gatekeeper.Spec.MutatingWebhook)
+
+	if validatingWebhookEnabled && mutatingWebhookEnabled {
```

Comment:
Why not starting with copy of orderedStaticAssets and removing under various conditions?
Like assets := copy()...
if !validatingWebhookEnabled { 
    // remove
}
return assets

I think it would make the function more linear

Reply:
Yes, this is one of those areas I wasn't completely happy with yet, but I decided to post the PR anyway due to time constraints. I originally started with your suggestion and opted for this after running into some golang issues. I will plan to clean up to make it easier to maintain.